pkg/trace/agent/agent.go

```diff
@@ -237,50 +237,41 @@ func (a *Agent) Process(t pb.Trace) {
 		pt.Env = tenv
 	}
 
-	go func() {
-		defer watchdog.LogOnPanic()
```

Comment:
I noticed that the new version of the code no longer has the `watchdog.LogOnPanic()` function to log panics.  I checked the code and these are all removed, I'm pretty sure we lose Panic logging.  I'd suggest adding a `defer watchdog.LogOnPanic()` at the top of the `Process()` func so that if any panics happen we get the logs.  If I'm wrong let me know.

Reply:
The `LogOnPanic` call would only catch panics for the goroutine where the defer statement is. Because we removed the goroutines these are not needed anymore. The goroutine running the Process method may already have a call to this function. If not, and needed, we can consider adding it (perhaps as the scope of another PR?).